### ISSUE TITLE
Sidebar is not reacting to navigation correctly

### ISSUE THREAD

**Issue by HymanZHAN** [angular/angular#57326]
### Describe the problem that you experienced

- When I was on the page "Common Routing Task" (https://angular.dev/guide/routing/common-router-tasks#lazy-loading), I clicked the "Lazy Loading" (https://angular.dev/guide/ngmodules/lazy-loading). 
- After navigation, the sidebar landed on the "References" part, not the most accurate "References/Concepts/NgModule/Lazy-loading feature modules"
- If I click backward, the sidebar will land on the top-level "Docs" part, not the more accurate "Docs/Routing/Common Routing Tasks"

### Enter the URL of the topic with the problem

https://angular.dev/guide/ngmodules/lazy-loading

### Add a screenshot if that helps illustrate the problem

https://github.com/user-attachments/assets/b0cccc4f-5cd3-4c7b-b3aa-88569cf8148c

**Comment by JeanMeche** [comment]
Hi, I was able to reproduce that specific issue. Could check if you see some error messages in the console, or maybe disable to see your extensions that could be messing with the site navigation ?

**Comment by HymanZHAN** [comment]
Nope, I have tried using different browsers, with or without plugins, and in private/incognito mode. The navigation behavior is the same.
The only error I see in the console:
![image](https://github.com/user-attachments/assets/3580b39e-b5c7-4466-9950-afc1cc6397a6)

**Comment by enstenr** [comment]
I tried replicating and found the same as specified. 
1. I do not see any error in the console
2. The side panel is not navigating to the correct location 
I observed the url 

Initiallly - > https://angular.dev/guide/routing/common-router-tasks#lazy-loading

After clicking lazy loading -> https://angular.dev/guide/ngmodules/lazy-loading

on click of back -> https://angular.dev/guide/routing/common-router-tasks#lazy-loading

the url is navigating correctly, so i guess this has to do something with the side panel.

**Comment by JeanMeche** [comment]
Could not reproduce after the recent updates. 

I someone can still reproduce it, please file a new issue.
